### PR TITLE
Add mine detector port from ACE2

### PR DESCRIPTION
**Pull request by Glowbal**
**When merged this pull request will:**
- Add mine detectors ported from ACE2
- Contains models from ACE2, work on models and animations done by @grey-soldierman

### CODE REVIEW

File: addons/minedetector/stringtable.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project name="ACE">
+    <Package name="MineDetector">
+        <Key ID="STR_ACE_MineDetector_">
```
Comment (bux578) [inline addons/minedetector/stringtable.xml]: ?

**Comment by Khaosmatic** [comment]
Awesome!

**Comment by commy2** [comment]
- [ ] CBA events
- [ ] remove unused sound files

File: addons/minedetector/XEH_clientInit.sqf [hunk 2]
```diff
@@ -0,0 +1,40 @@
+#include "script_component.hpp"
+
+
+[QGVAR(detectorEnabled), {
```
Comment (jonpas) [inline addons/minedetector/XEH_clientInit.sqf]: Those 2 events are only called locally, why are they events in the first place? Either use direction function call or make them public?

File: addons/minedetector/config.cpp [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#include "script_component.hpp"
+
+class CfgPatches {
+    class ADDON {
+        units[] = {};
+        weapons[] = {"ACE_VMH3","ACE_VMM3"};
+        requiredVersion = REQUIRED_VERSION;
+        requiredAddons[] = {"ace_explosives"};
+        author[] = {"Grey", "Glowbal", "Rocko"};
```
Comment (jonpas) [inline addons/minedetector/config.cpp]: `author = ECSTRING(common,ACETeam`
`authors[]` (S)
`url`

File: addons/minedetector/data/anim/model.cfg [hunk 4]
```diff
@@ -0,0 +1,154 @@
+class CfgSkeletons
+{
```
Comment (jonpas) [inline addons/minedetector/data/anim/model.cfg]: This file doesn't fit our coding guidelines (brace on same line):

**Comment by Omnirock** [comment]
Greatly anticipating the arrival of mine detection equipment. Additionally, having the ability to mark said mines would be a brilliant addition to anyone responsible for detecting them. Being able to place a flag or marker of some sort physically on the ground would be a great supplement to this.

**Comment by Glowbal** [comment]
Will create a seperate issue for the icons.

File: addons/minedetector/functions/fnc_playDetectorSound.sqf [hunk 5]
```diff
@@ -34,6 +34,8 @@ _helperObject = "ACE_LogicDummy" createVehicleLocal (getPos _unit);
 
 if !(isNull _helperObject) then {
     _helperObject attachto [_unit,[0,0,-3],""];
+    _unit setvariable [QGVAR(helperLogic), _helperObject];
```
Comment (jonpas) [inline addons/minedetector/functions/fnc_playDetectorSound.sqf]: That capitalization triggers me. :D

Comment (commy2) [inline addons/minedetector/functions/fnc_playDetectorSound.sqf]: ACE Code guidelines say to always use the camel case for SQF commands.

Comment (bux578) [inline addons/minedetector/functions/fnc_playDetectorSound.sqf]: fixed in: https://github.com/acemod/ACE3/pull/3936/commits/2d029401f5e05ccdce34259fcdb256b36f28926a

**Pull request merged**